kurtosis/src/lib/service_config.star

```diff
@@ -131,12 +133,23 @@ def validate_service_config_types(service_config):
     # TODO(validation): Implement validation for tolerations
     # TODO(validation): Implement validation for node_selectors
 
-def create_from_config(config):
+def create_public_port_specs_from_config(config, is_full_node):
+    ports = {}
+    if is_full_node:
+        ports = {}
+        for port_key, port_spec in config["public_ports"].items():
+            ports[port_key] = port_spec_lib.create_port_spec(port_spec)
+
+    return ports
```

Comment:
Does otterscan let you customize the ETH JSON-RPC port you're hitting? It seems messy to put in all these exceptions just to expose erigon's RPC as 8545 when we can pass along the generated port like we do with other services

Reply:
I would prefer to leave 8545 available on local machine because you never know when someone is already running a local node on 8545. For example, I commonly port-forward a kubernetes node to check things on another network and will usually use 8545 when doing so

Reply:
Otterscan uses ERIGON_URL to customize the URL so yes we could change the port. As otterscan communicates with the erigon RPC URL, port 8545 does not work in this case unlike other services, we need the public port that is exposed by kurtosis and since kurtosis does it in random order there is no way to figure out the public port from ServiceConfig. So I am explicitly setting the public port for the erigon full node, I kept it the same for ease of identification. We could change the public port to any other port, may be 8546 or others ?
I agree with the point whether 8545 might be occupied on the machine.

Reply:
nidhi tried all the other variants, seems the hostname : port method isn't working with erigon for some reason so marking this as resolved